core/merginuserinfo.h

```diff
@@ -24,28 +24,42 @@ class MerginUserInfo: public QObject
     Q_PROPERTY( QString email READ email NOTIFY userInfoChanged )
     Q_PROPERTY( double storageLimit READ storageLimit NOTIFY userInfoChanged )
     Q_PROPERTY( double diskUsage READ diskUsage NOTIFY userInfoChanged ) // in Bytes
+    Q_PROPERTY( QString activeWorkspace READ activeWorkspace NOTIFY userInfoChanged )
+    Q_PROPERTY( QStringList workspaces READ workspaces NOTIFY userInfoChanged )
 
   public:
     explicit MerginUserInfo( QObject *parent = nullptr );
     ~MerginUserInfo() = default;
 
-  public:
     void clear();
     void setFromJson( QJsonObject docObj );
 
     QString email() const;
     double diskUsage() const;
     double storageLimit() const;
+    QString activeWorkspace() const;
+    QStringList workspaces() const;
+
+    void saveWorkspacesData();
+    void loadWorkspacesData();
+
+    void findActiveWorkspace();
+    void saveLastActiveWorkspace();
 
   signals:
     void userInfoChanged();
+
   public slots:
     void onStorageChanged( double storage );
 
   private:
     QString mEmail;
     double mDiskUsage = 0.0; // in Bytes
     double mStorageLimit = 0.0; // in Bytes
+    int mPreferredWorkspace = -1;
```

Comment:
I would also not cache, nor store the `preferredWorkspace` here, it is used only once per user lifetime - can be simply just parsed and used :)

Reply:
Done.